### PR TITLE
Cleaning and Fixing public API for set of learners.

### PR DESCRIPTION
**Pull request by zeahmed**
This PR partially addressed #2620

The following learners are addressed in this PR.
* `MatrixFactorizationTrainer`
* `PriorTrainer`
* `RandomTrainer`
* `SymSgdClassificationTrainer`

the following tasks were performed in classes related to above learners.

* Checking to make sure that unnecessary public methods/properties be `internal`.
* Renaming parameters according to standard.
* Creating/Refactoring samples according to standards.

### CODE REVIEW

File: src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs [hunk 1]
```diff
@@ -223,7 +223,7 @@ private protected override BinaryPredictionTransformer<TPredictor> MakeTransform
              => new BinaryPredictionTransformer<TPredictor>(Host, model, trainSchema, FeatureColumn.Name);
 
         /// <summary>
-        /// Continues the training of a <see cref="SymSgdClassificationTrainer"/> using an already trained <paramref name="modelParameters"/>
+        /// Continues the training of a <see cref="SymbolicStochasticGradientDescentClassificationTrainer"/> using an already trained <paramref name="modelParameters"/>
         /// a <see cref="BinaryPredictionTransformer"/>.
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.HalLearners/SymSgdClassificationTrainer.cs]: what `a` means in this sentence?

File: src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs [hunk 2]
```diff
@@ -52,12 +52,16 @@ private static VersionInfo GetVersionInfo()
...
         ///<summary> The number of rows.</summary>
-        public readonly int NumberOfRows;
+        public int NumberOfRows { get; }
+
         ///<summary> The number of columns.</summary>
-        public readonly int NumberOfColumns;
+        public int NumberOfColumns { get; }
+
         ///<summary> The rank of the factor matrices.</summary>
-        public readonly int ApproximationRank;
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Recommender/MatrixFactorizationPredictor.cs]: >ApproximationRank [](start = 28, length = 17)

I'm not sure this one is any way better than previous code. Now you need to initialize new variable.

**Comment by Ivanidzo4ka** [comment]
>         public static OlsLinearRegressionTrainer OrdinaryLeastSquares(this RegressionCatalog.RegressionTrainers catalog,

OrdinaryLeastSquaresRegressionTrainer?

---
Refers to: src/Microsoft.ML.HalLearners/HalLearnersCatalog.cs:30 in 8235b0a. [](commit_id = 8235b0a4106057222b64b4eda40c5a215b614c9f, deletion_comment = False)

File: src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs [hunk 3]
```diff
@@ -100,8 +100,8 @@ public static IDataView[] LoadFeaturizedSentimentDataset(MLContext mlContext)
             // Download the files
             var dataFiles = DownloadSentimentDataset();
 
-            // Define the columns to load
-            var loader = mlContext.Data.CreateTextLoader(
+            // Define the columns to read
+            var reader = mlContext.Data.CreateTextLoader(
```
Comment (sfilipi) [inline src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: >reader [](start = 16, length = 6)

nit: if the Name is CreateTextLoader let;s stick with Load.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/PriorTrainerSample.cs [hunk 4]
```diff
@@ -7,59 +7,64 @@ public class PriorTrainer
...
-            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
-            // as a catalog of available operations and as the source of randomness.
-            var mlContext = new MLContext();
-
-            // Step 1: Load the data as an IDataView.
-            // First, we define the loader: specify the data columns and where to find them in the text file.
-            var loader = mlContext.Data.CreateTextLoader(
+            // Step 1: Read the data as an IDataView.
+            // First, we define the reader: specify the data columns and where to find them in the text file.
+            var reader = mlContext.Data.CreateTextLoader(
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/PriorTrainerSample.cs]: >reader [](start = 16, length = 6)

load. Artidoro had a whole PR to replace Read with Load.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/PriorTrainerSample.cs [hunk 5]
```diff
@@ -7,59 +7,64 @@ public class PriorTrainer
...
-            var trainTestData = mlContext.BinaryClassification.TrainTestSplit(data);
+            // Read the data
+            var trainData = reader.Load(trainFile);
 
             // Step 2: Pipeline 
             // Featurize the text column through the FeaturizeText API. 
             // Then append a binary classifier, setting the "Label" column as the label of the dataset, and 
             // the "Features" column produced by FeaturizeText as the features column.
             var pipeline = mlContext.Transforms.Text.FeaturizeText("Features", "SentimentText")
-                    .AppendCacheCheckpoint(mlContext) // Add a data-cache step within a pipeline.
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/PriorTrainerSample.cs]: >// Add a data-cache step within a pipeline. [](start = 54, length = 43)

why remove it?

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/RandomTrainerSample.cs [hunk 6]
```diff
@@ -7,59 +7,64 @@ public static class RandomTrainer
...
                 hasHeader: true
             );
-            
-            // Load the data
-            var data = loader.Load(dataFile);
 
-            // Split it between training and test data
-            var trainTestData = mlContext.BinaryClassification.TrainTestSplit(data);
+            // Read the data
+            var trainData = reader.Load(trainFile);
```
Comment (sfilipi) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/RandomTrainerSample.cs]: >reader.Load [](start = 28, length = 11)

nit.

**Comment by Ivanidzo4ka** [comment]
>             public float L2Weight = 1e-6f;

See my comments in your other PR which also touch OLS.
I'm fine with pushing this one first, just make sure you address comments in other PR.

---
Refers to: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs:50 in 88a14c4. [](commit_id = 88a14c4dc0bc6fc211a465236da22080957fbbf2, deletion_comment = False)

File: src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs (hunk 3 above)
Comment (Ivanidzo4ka) [reply src/Microsoft.ML.SamplesUtils/SamplesDatasetUtils.cs]: Still reader.

---
In reply to: [261714176](https://github.com/dotnet/machinelearning/pull/2765#discussion_r261714176) [](ancestors = 261714176)

**Comment by zeahmed** [comment]
>             public float L2Weight = 1e-6f;

OK, I will fix that in the other PR.

---
In reply to: [468847316](https://github.com/dotnet/machinelearning/pull/2765#issuecomment-468847316) [](ancestors = 468847316)

---
Refers to: src/Microsoft.ML.HalLearners/OlsLinearRegression.cs:50 in 88a14c4. [](commit_id = 88a14c4dc0bc6fc211a465236da22080957fbbf2, deletion_comment = False)

**Pull request merged**